content/guides/identity-management/quick-start/5-find-or-create-box-users.md

```diff
@@ -0,0 +1,499 @@
+
+Replace the `// TODO: VALIDATE USER` comment with the following:
+
+```java
+  // Set up Box enterprise client
+  Reader reader = new FileReader("config.json");
+  BoxConfig config = BoxConfig.readFrom(reader);
+  api = BoxDeveloperEditionAPIConnection.getAppEnterpriseConnection(config);
+
+  // Get Okta user sub for unique ID attachment to Box user
+  Object oktaSub = user.getAttributes().get("sub");
+
+  // Check enterprise users for matching external_app_user_id against Okta sub
+  URL url = new URL("https://api.box.com/2.0/users?external_app_user_id=" + oktaSub);
+  BoxAPIRequest request = new BoxAPIRequest(api, url, "GET");
+  BoxJSONResponse jsonResponse = (BoxJSONResponse) request.send();
+  JsonObject jsonObj = jsonResponse.getJsonObject();
+  JsonValue totalCount = jsonObj.get("total_count");
+
+  // Set return string
+  String outputString = "";
+
+  if (totalCount.asInt() > 0) {
+    // TODO: MAKE AUTHENTICATED USER CALL
+  } else {
+    outputString = createUser(user);
+  }
+
+  return outputString;
+```
+
+Using the Box Java SDK generic request method, we make a call directly to the
+`https://api.box.com/2.0/users` endpoint to search enterprise users, passing in
+the unique Okta user ID as the `external_app_user_id` value to search
+specifically for that user.
+
+If found (number of records is greater than 0) we can make an
+authenticated call to the Box APIs using that user record, which will be
+defined in the next section. 
+
+If not found, we call the `createUser` function we defined in the last section
+to create a new Box user with that `external_app_user_id` association.
+
+</Choice>
+<Choice option='programming.platform' value='python' color='none'>
+
+Replace the `# TODO: VALIDATE USER` comment with the following:
+
+```python
+  # Fetch Okta user ID
+  uid = g.user.id
+
+  # Validate is user exists
+  url = f'https://api.box.com/2.0/users?external_app_user_id={uid}'
+  response = self.box_client.make_request('GET', url)
+  user_info = response.json()
+
+  # If user not found, create user, otherwise fetch user token
+  if (user_info['total_count'] == 0):
+    self.createUser(g.user)
+  else:
+    # TODO: MAKE AUTHENTICATED USER CALL
+
+  return 'Complete'
+```
+
+Using the Box Python SDK generic request method, we make a call directly to the
+`https://api.box.com/2.0/users` endpoint to search enterprise users, passing in
+the unique Okta user ID as the `external_app_user_id` value to search
+specifically for that user.
+
+If found (number of records is greater than 0) we can make an
+authenticated call to the Box APIs using that user record, which will be
+defined in the next section. 
+
+If not found, we call the `createUser` function we defined in the last section
+to create a new Box user with that `external_app_user_id` association.
+
+</Choice>
+<Choice option='programming.platform' value='node' color='none'>
+
+Replace the `// TODO: VALIDATE USER` comment with the following:
+
+```cs
```

Comment:
The code for `cs` is `dotnet`. I can add this into the codebase though, so it will map right.

Reply:
Adjusted all cs / csharp instances to use a uniform dotnet.